R/wrangle_and_check.R

```diff
@@ -15,9 +15,22 @@ wrangle_and_check_sector_exposures <- function(sector_exposures, asset_type) {
   asset_type <- stringr::str_to_title(asset_type)
 
   if (!asset_type %in% c("Bonds", "Equity", "Loans")) {
-    stop("Invalid asset type")
+    stop("Invalid asset type", call. = FALSE )
   }
 
+  if (!is.logical(sector_exposures$valid_input) ||
+      !is.character(sector_exposures$financial_sector) ||
+      any(sector_exposures$valid_value_usd < 0)) {
+    stop("Detected implausible inputs on sector exposures", call. = FALSE)
+  }
+
+  check_data_structure(
+    data = sector_exposures,
+    name_data = "sector exposures",
+    cuc_cols = "financial_sector",
```

Comment:
I think I disagree with this selection of CUC cols here. Made a comment in the data dictionary https://docs.google.com/spreadsheets/d/17VU49YwfIkv82dkFIg5EyZinp_A7EetlR9LrXP8eNlU/edit#gid=214413387

Reply:
you are so right, and I should have noticed by testing all asset_types, eq and cb would fail with the current cuc, fixed.